pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -222,7 +225,17 @@ func (c *objectCache) newReflectorLocked(namespace, name string) *objectCacheIte
 		options.FieldSelector = fieldSelector
 		return c.watchObject(namespace, options)
 	}
-	store := c.newStore()
+	refPods := map[types.UID]int{}
+	pushFunc := func([]interface{}) {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+		for podUID := range refPods {
+			c.podsNeedSyncObject.Insert(podUID)
```

Comment:
this closure taking objectCache-level locks and modifying a local map is pretty hard to reason about ... is there a way we can layer in managing item → pod updates without making the watch_based_manager pod-specific or embedding state in closures like this?